lib/Boltpay/Bugsnag/Notification.php

```diff
@@ -171,16 +171,13 @@ public function postJSON($url, $data)
             return;
         }
 
-        // Prefer cURL if it is installed, otherwise fall back to fopen()
-        // cURL supports both timeouts and proxies
-
         try {
-            if (function_exists('curl_version')) {
-                $this->postWithCurl($url, $body);
+            if (method_exists('GuzzleHttp\Client','sendAsync')) {
```

Comment:
Since you are choosing to modify the 3rd party Bugsnag library by removing the `postWithCurl`, which is the aggressive approach, there are a few observations to point out.  

First, since we are including guzzle with our software, then Guzzle must exist, right?  That would make this check unnecessary.  We should always `postWithGuzzle`.

Second, internally, guzzle is performing the same check to see if `curl_multi_exec` and `curl_exec` exists and whether `ini_get('allow_url_fopen')` is set to true.  Therefore, `postWithFopen` would also be unnecessary because Guzzle will do that internally.

https://github.com/guzzle/guzzle/blob/3b0452a3a0347e74c42e9b9a2e9b11158531ce14/src/functions.php#L95-L124

You could also choose to take the more conservative approach which would be to check for `method_exists('GuzzleHttp\Client','sendAsync')` to exist and use it, but keeping the original cascade intact by then checking for `function_exists('curl_version')` and not removing `postWithCurl` or the next check for  `ini_get('allow_url_fopen')` and `postWithFopen`